ui/app/mixins/searchable.js

```diff
@@ -17,17 +25,77 @@ import { get, computed } from '@ember/object';
 export default Mixin.create({
   searchTerm: '',
   listToSearch: computed(() => []),
+
   searchProps: null,
+  exactMatchSearchProps: reads('searchProps'),
+  regexSearchProps: reads('searchProps'),
+  fuzzySearchProps: reads('searchProps'),
 
-  listSearched: computed('searchTerm', 'listToSearch.[]', 'searchProps.[]', function() {
-    const searchTerm = this.get('searchTerm');
-    if (searchTerm && searchTerm.length) {
-      return regexSearch(searchTerm, this.get('listToSearch'), this.get('searchProps'));
-    }
-    return this.get('listToSearch');
+  // Three search modes
+  exactMatchEnabled: true,
+  fuzzySearchEnabled: false,
+  regexEnabled: true,
+
+  fuse: computed('listToSearch.[]', 'fuzzySearchProps.[]', function() {
+    return new Fuse(this.get('listToSearch'), {
+      shouldSort: true,
+      threshold: 0.4,
+      location: 0,
+      distance: 100,
+      tokenize: true,
+      matchAllTokens: true,
+      maxPatternLength: 32,
+      minMatchCharLength: 1,
+      keys: this.get('fuzzySearchProps') || [],
+      getFn(item, key) {
+        return get(item, key);
+      },
+    });
   }),
+
+  listSearched: computed(
+    'searchTerm',
+    'listToSearch.[]',
+    'exactMatchEnabled',
+    'fuzzySearchEnabled',
+    'regexEnabled',
+    'exactMatchSearchProps.[]',
+    'fuzzySearchProps.[]',
+    'regexSearchProps.[]',
+    function() {
+      const searchTerm = this.get('searchTerm').trim();
+      if (searchTerm && searchTerm.length) {
+        const results = [];
+        if (this.get('exactMatchEnabled')) {
+          results.push(
+            ...exactMatchSearch(
+              searchTerm,
+              this.get('listToSearch'),
+              this.get('exactMatchSearchProps')
+            )
+          );
+        }
+        if (this.get('fuzzySearchEnabled')) {
+          results.push(...this.get('fuse').search(searchTerm));
+        }
+        if (this.get('regexEnabled')) {
+          results.push(
+            ...regexSearch(searchTerm, this.get('listToSearch'), this.get('regexSearchProps'))
+          );
+        }
+        return results.uniq();
+      }
+      return this.get('listToSearch');
```

Comment:
Inverting the condition and returning the full list easier would reduce indentation for the search code and might be a little clearer.